File: 251109/숫자가 더 큰 인접한 곳으로 이동/move-to-larger-adjacent-cell.py
n, r, c = map(int, input().split())
a = [[0] * (n + 1) for _ in range(n + 1)]

for i in range(1, n + 1):
    row = list(map(int, input().split()))
    for j in range(1, n + 1):
        a[i][j] = row[j - 1]

# Please write your code here.
d = [(-1,0),(1,0),(0,-1),(0,1)] #상하좌우

move = 1
record = [a[r][c]]
while move:
    for dr, dc in d:
        nr = r + dr
        nc = c + dc
        if 0<= nr < n and 0 <= nc < n and a[nr][nc] > a[r][c]:
            record.append(a[nr][nc])
            r = nr
            c = nc
            break
    else:
        move = 0

print(*record)

File: 260113/숫자가 더 큰 인접한 곳으로 이동/move-to-larger-adjacent-cell.java
import java.util.*;
public class Main {
    static int[][] grid;

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int r = sc.nextInt();
        int c = sc.nextInt();
        grid = new int[n+1][n+1];
        for (int i = 1; i <= n; i++) {
            for (int j = 1; j <= n; j++) {
                grid[i][j] = sc.nextInt();
            }
        }
        // Please write your code here.
        int[][] d = {{-1,0},{1,0},{0,-1},{0,1}};
        ArrayDeque<int[]> q = new ArrayDeque<>();
        boolean[][] visited = new boolean [n+1][n+1];
        q.add(new int[] {r,c});
        visited[r][c] = true;
        System.out.print(grid[r][c]+" ");

        while(!q.isEmpty()){
            int[] t = q.pollFirst();
            for(int[] v : d){
                int nr = t[0] + v[0];
                int nc = t[1] + v[1];
                if(0<=nr && nr<n && 0<=nc && nc<n && !visited[nr][nc] && (grid[t[0]][t[1]] < grid[nr][nc])){
                    System.out.print(grid[nr][nc]+" ");
                    q.add(new int[] {nr,nc});
                    visited[nr][nc] = true;
                    break;
                }
            }
        }
    }
}